nucypher/network/nodes.py

```diff
@@ -985,9 +979,19 @@ class UnbondedWorker(InvalidNode):
     class WrongMode(TypeError):
         """Raised when a Character tries to use another Character as decentralized when the latter is federated_only."""
 
-    class IsFromTheFuture(TypeError):
+    class UnexpectedVersion(TypeError):
+        """Raised when deserializing a Character from a unexpected and incompatible version."""
+
+    class IsFromTheFuture(UnexpectedVersion):
         """Raised when deserializing a Character from a future version."""
 
+    class AreYouFromThePast(UnexpectedVersion):
+        """Raised when deserializing a Character from a previous, now unsupported version."""
+
+    unknown_version_message = "{} purported to be of version {}, but we're version {}."
+    really_unknown_version_message = "Unable to glean address from node that perhaps purported to be version {}. " \
```

Comment:
"perhaps purported" feels redundant.

```suggestion
    really_unknown_version_message = "Unable to glean address from node that purported to be version {}. " \
```

Reply:
✅